props/saber_BC_buttons.h

```diff
@@ -386,31 +1134,317 @@ EFFECT(volmax);     // for maximum volume reached
 #warning The faston.wav sound will be replaced with fastout.wav. If you have a good reason to keep faston.wav as is, please post at https://crucible.hubbe.net/
 EFFECT(faston);     // for EFFECT_FAST_ON. Being replaced by fastout.wav, which is already defined in the main OS.
 #endif
-EFFECT(blstbgn);    // for Begin Multi-Blast
-EFFECT(blstend);    // for End Multi-Blast
 EFFECT(push);       // for Force Push gesture
-EFFECT(monosfx);    // for Monophonically played sounds (iceblade, seismic charge etc...)
-EFFECT(swap);       // for standalone triggering EffectSequence<>
+EFFECT(tr);         // for EFFECT_TRANSITION_SOUND, use with User Effects.
+EFFECT2(trloop, trloop);  // have YET to play with this
 EFFECT(mute);       // Notification before muted ignition to avoid confusion.
+EFFECT(mzoom);      // for Spam Blast enter/exit
+
+#ifndef BC_ENABLE_OS_MENU
+template<class SPEC>
+struct BCVolumeMode : public SPEC::MenuBase {
+  const int max_volume_ = VOLUME;
+  const int min_volume_ = VOLUME * 0.10;
+  float initial_volume_ = 0.0;
+  int initial_percentage_ = 0;
+  int last_volume_ = 0;
+  int current_volume_ = 0;
+  int percentage_ = 0;
+
+int steps_per_revolution() override {
+  return 12;  // adjust for sensitivity
+}
+  uint16_t size() override {
+    return 10;
+  }
 
-class DelayTimer {
-public:
-    DelayTimer() : triggered_(false), trigger_time_(0), duration_(0) {}
+  void mode_activate(bool onreturn) override {
+    PVLOG_NORMAL << "** Enter Volume Menu\n";
+    initial_volume_ = dynamic_mixer.get_volume();
+    initial_percentage_ = round((initial_volume_ / max_volume_) * 10) * 10;
+    last_volume_ = initial_volume_;
+    current_volume_ = initial_volume_;
+   SaberBase::DoEffect(EFFECT_VOLUME_LEVEL, 0);
+    mode::getSL<SPEC>()->SayEnterVolumeMenu();
+    SPEC::SteppedMode::mode_activate(onreturn);
+  }
 
-    void trigger(uint32_t duration) {
-        triggered_ = true;
-        trigger_time_ = millis();
-        duration_ = duration;
+  void mode_deactivate() override {
+    if (cancelling_) {
+      percentage_ = initial_percentage_;
+      dynamic_mixer.set_volume(initial_volume_);
     }
+    if (percentage_ <= 10) {
+      mode::getSL<SPEC>()->SayMinimumVolume();
+    } else if (percentage_ >=100) {
+      mode::getSL<SPEC>()->SayMaximumVolume();
+    } else {
+      mode::getSL<SPEC>()->SayWhole(percentage_);
+      mode::getSL<SPEC>()->SayPercent();
+    }
+    mode::getSL<SPEC>()->SayVolumeMenuEnd();
+    SPEC::MenuBase::mode_deactivate();
+  }
 
-    bool timerCheck() {
-        if (!triggered_) return false;
-        if (millis() - trigger_time_ > duration_) {
-            triggered_ = false;  // Reset the timer flag
-            return true;  // Timer has elapsed
-        }
-        return false;  // Timer is still running
+  void say() override {
+    if (last_volume_ < current_volume_) {
+      mode::getSL<SPEC>()->SayVolumeUp();
```

Comment:
Why not do SayVolumeUp() in next() and SayVolumeDown() in Prev() ?

Reply:
I wanted to, but they only speak if they are in say(). I do not understand why but nonetheless that is the case.

Reply:
Hmm, I guess that's because there is a fadeout() in update(), which is called right after next/prev.
I guess even the SteppedMode assumes that say() will announce the current state rather than speak to what action was taken.

It feels like the right thing to do here would be to split SteppedMode into two classes, something like SteppedModeBase and SteppedMode. SteppedModeBase would just provide next() prev(), but not make any assumptions about the sound, while SteppedMode would add update(), say() and all the other stuff that SteppedMode does....

Although, the current implementation works too I suppose...

Reply:
That's fine since I'm overriding update() anyway so SayVolumeUp and Down can be used instead of ccchange, and there's no speaking at each step anyway, so by removing fadeout in this case allows speaking in next() and prev() tp work fine.

Reply:
done